src/lexer/lex.c

```diff
@@ -24,3 +24,110 @@ int unlex(Lexer *l, Token *t) {
     l->unlexed_count++;
     return 0;
 }
+
+TokenType ttype_one_char(char c) {
+    switch (c) {
+    case '(':
+        return TT_OPAREN; // (
+    case ')':
+        return TT_CPAREN; // )
+    case '{':
+        return TT_OBRACE; // {
+    case '}':
+        return TT_CBRACE; // }
+    case '[':
+        return TT_OBRACKET; // [
+    case ']':
+        return TT_CBRACKET; // ]
+    case ';':
+        return TT_SEMI; // ;
+    }
+
+    return TT_NO_TOKEN;
+}
+
+TokenType ttype_many_chars(const char *contents) {
+    // Includes only numbers
+    int all_numeric = 1;
+
+    int count_fs = 0;
+    int count_us = 0;
+
+    int i;
+
+    if (contents == NULL) {
+        return TT_NO_TOKEN;
+    }
+
+    // Does it start with a negative sign
+    if (contents[0] != '\0') {
+        if (contents[0] == '-') {
+            return TT_IDENTIFIER;
+        }
+    }
+
+    // Loop through each character
+    for (i = 0; contents[i] != '\0'; i++) {
+        char c = contents[i];
+
+        // If it has a period, it's a float
+        if (c == '.') {
+            return TT_LITERAL;
+        }
+
+        // Count negative signs
+        if (c == '-') {
+            return TT_LITERAL;
+        }
+
+        // Count 'f'
+        if (c == 'f') {
+            count_fs++;
+        }
+
+        // Count 'u'
+        if (c == 'u') {
+            count_us++;
+        }
+
+        // Is it from "0123456789"?
+        if (c > 57 || c < 48 && c != 'u') {
```

Comment:
I might use the literal characters here instead of `57` and `48` so it's clearer

Reply:
Yup